types/mapcoins_test.go

```diff
@@ -0,0 +1,63 @@
+package types_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func (s *coinTestSuite) TestMapCoinsAdd() {
+	cA0M0 := sdk.Coins{s.ca0, s.cm0}
+	cA0M1 := sdk.Coins{s.ca0, s.cm1}
+	cA1M1 := sdk.Coins{s.ca1, s.cm1}
+	cases := []struct {
+		name     string
+		inputOne sdk.Coins
+		inputTwo sdk.Coins
+		expected sdk.Coins
+		msg      string
+	}{
+		{"adding two empty lists", s.emptyCoins, s.emptyCoins, s.emptyCoins, "empty, non list should be returned"},
+		{"empty list + set", s.emptyCoins, cA0M1, sdk.Coins{s.cm1}, "zero coins should be removed"},
+		{"empty list + set", s.emptyCoins, cA1M1, cA1M1, "zero + a_non_zero = a_non_zero"},
```

Comment:
These two names should be different to clarify that the earlier case adds a zero coin and the later case doesn't.

Same for the next two cases.

Reply:
These are copy pasted from the sdk Coins add tests